dist/js/script.js

```diff
@@ -325,7 +378,51 @@
       });
 
     }
+
+    add(menuProduct){
+      const thisCart = this;
+      console.log('adding product', menuProduct);
+  
+      /*generare HTML based on template */
+      const generateHTML = templates.cartProduct(menuProduct);
+  
+      /*create element using utils.createElementFromHTML */
+      menuProduct.element = utils.createDOMFromHTML(generateHTML);
+   
+      /*add elemnt to menu */
+      thisCart.dom.productList.appendChild(menuProduct.element);
+
+      thisCart.products.push(new CartProduct(menuProduct, generatedDOM));
+      console.log('thisCart.products', thisCart.products);
+      }
+    }
```

Comment:
W metodzie `add` coś chyba pomieszałaś ;) według instrukcji  zadania wyglada prościej. Poza tym nie widzę powodu, żeby w jakikolwiek sposób modyfikować wartość z argumentu `menuProduct`. Spróbuj tak:
```suggestion
    add(menuProduct) {
      const thisCart = this

      /* generate HTML based on template */
      const generatedHTML = templates.cartProduct(menuProduct)

      /* create element using utils.createDOMFromHTML */
      const generatedDOM = utils.createDOMFromHTML(generatedHTML)

      /* add element to menu */
      thisCart.dom.productList.appendChild(generatedDOM)
      thisCart.products.push(new CartProduct(menuProduct, generatedDOM))
    }
    }
```